server/src/repositories/partner.repository.ts

```diff
@@ -1,37 +1,93 @@
 import { Injectable } from '@nestjs/common';
-import { InjectRepository } from '@nestjs/typeorm';
+import { ExpressionBuilder, Insertable, JoinBuilder, Kysely, Updateable } from 'kysely';
+import { jsonObjectFrom } from 'kysely/helpers/postgres';
+import { InjectKysely } from 'nestjs-kysely';
+import { DB, Partners, Users } from 'src/db';
+import { DummyValue, GenerateSql } from 'src/decorators';
 import { PartnerEntity } from 'src/entities/partner.entity';
 import { IPartnerRepository, PartnerIds } from 'src/interfaces/partner.interface';
-import { DeepPartial, Repository } from 'typeorm';
+
+const columns = ['id', 'name', 'email', 'profileImagePath', 'profileChangedAt'] as const;
+
+const onSharedBy = (join: JoinBuilder<DB & { sharedBy: Users }, 'partners' | 'sharedBy'>) =>
+  join.onRef('partners.sharedById', '=', 'sharedBy.id').on('sharedBy.deletedAt', 'is', null);
+
+const onSharedWith = (join: JoinBuilder<DB & { sharedWith: Users }, 'partners' | 'sharedWith'>) =>
+  join.onRef('partners.sharedWithId', '=', 'sharedWith.id').on('sharedWith.deletedAt', 'is', null);
+
+const withSharedBy = (eb: ExpressionBuilder<DB, 'partners'>) => {
+  return jsonObjectFrom(
+    eb.selectFrom('users as sharedBy').select(columns).whereRef('sharedBy.id', '=', 'partners.sharedById'),
+  ).as('sharedBy');
+};
+
+const withSharedWith = (eb: ExpressionBuilder<DB, 'partners'>) => {
+  return jsonObjectFrom(
+    eb.selectFrom('users as sharedWith').select(columns).whereRef('sharedWith.id', '=', 'partners.sharedWithId'),
+  ).as('sharedWith');
+};
 
 @Injectable()
 export class PartnerRepository implements IPartnerRepository {
-  constructor(@InjectRepository(PartnerEntity) private repository: Repository<PartnerEntity>) {}
+  constructor(@InjectKysely() private db: Kysely<DB>) {}
 
+  @GenerateSql({ params: [DummyValue.UUID] })
   getAll(userId: string): Promise<PartnerEntity[]> {
-    return this.repository.find({ where: [{ sharedWithId: userId }, { sharedById: userId }] });
-  }
-
-  get({ sharedWithId, sharedById }: PartnerIds): Promise<PartnerEntity | null> {
-    return this.repository.findOne({ where: { sharedById, sharedWithId } });
+    return this.db
+      .selectFrom('partners')
+      .innerJoin('users as sharedBy', onSharedBy)
+      .innerJoin('users as sharedWith', onSharedWith)
+      .selectAll('partners')
+      .select(withSharedBy)
+      .select(withSharedWith)
```

Comment:
```suggestion
      .select((eb) => jsonObjectFrom(eb.selectFrom('sharedBy').select(columns)).as('sharedBy'))
      .select((eb) => jsonObjectFrom(eb.selectFrom('sharedWith').select(columns)).as('sharedWith'))
```
`withSharedBy`/`withSharedWith` effectively repeat the work that the inner joins do, so it's better to select from the already-joined tables.

Reply:
![image](https://github.com/user-attachments/assets/73e878dc-ed3b-42a4-96de-903009fc2b3e)